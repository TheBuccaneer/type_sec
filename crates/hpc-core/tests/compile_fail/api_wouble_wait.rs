use hpc_core::*;

fn main() {
    let ctx = Context::create_context().unwrap();
    let queue = ctx.create_queue().unwrap();
    
    // Buffer → Ready
    let buf = ctx
        .create_buffer::<u8>(16).unwrap()
        .enqueue_write(&queue, &[0u8; 16]).unwrap();
    
    // Non-blocking write statt kernel (einfacher für Test)
    let (inflight, evt) = buf.overwrite_non_blocking(&queue, &[1u8; 16]).unwrap();
    
    // Erster Wait (legal)
    let ready = evt.wait(inflight);
    
    // Zweiter Wait (illegal, evt wurde schon konsumiert)
    let _illegal = evt.wait(ready);
    //             ^^^ use of moved value `evt`
}
