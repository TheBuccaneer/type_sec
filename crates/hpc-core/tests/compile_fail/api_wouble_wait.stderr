error[E0599]: no method named `unwrap` found for struct `DeviceBuffer` in the current scope
  --> tests/compile_fail/api_wouble_wait.rs:16:36
   |
16 |     let ready = evt.wait(inflight).unwrap();
   |                                    ^^^^^^ method not found in `DeviceBuffer<'_, u8, hpc_core::Ready>`

error[E0599]: no method named `unwrap` found for struct `DeviceBuffer` in the current scope
  --> tests/compile_fail/api_wouble_wait.rs:19:36
   |
19 |     let _illegal = evt.wait(ready).unwrap();
   |                                    ^^^^^^ method not found in `DeviceBuffer<'_, _, hpc_core::Ready>`
